nslsii/__init__.py

```diff
@@ -336,28 +342,30 @@ def configure_bluesky_logging(ipython, appdirs_appname="bluesky"):
         "  %(module)s:%(lineno)d] %(message)s"
     )
     log_file_handler.setFormatter(logging.Formatter(fmt=log_file_format))
+
     logging.getLogger("bluesky").addHandler(log_file_handler)
     logging.getLogger("caproto").addHandler(log_file_handler)
     logging.getLogger("ophyd").addHandler(log_file_handler)
     logging.getLogger("nslsii").addHandler(log_file_handler)
-    if ipython:
-        ipython.log.addHandler(log_file_handler)
+
     # set the loggers to send INFO and higher log
     # messages to their handlers
     logging.getLogger("bluesky").setLevel("INFO")
     logging.getLogger("caproto").setLevel("INFO")
     logging.getLogger("ophyd").setLevel("INFO")
     logging.getLogger("nslsii").setLevel("INFO")
-    if ipython:
-        ipython.log.setLevel("INFO")
+
     # configure loggers so that log messages do not
     # propagate to higher level loggers
-    logging.getLogger("bluesky").propagate = False
-    logging.getLogger("caproto").propagate = False
-    logging.getLogger("ophyd").propagate = False
-    logging.getLogger("nslsii").propagate = False
+    logging.getLogger("bluesky").propagate = propagate_log_messages
+    logging.getLogger("caproto").propagate = propagate_log_messages
+    logging.getLogger("ophyd").propagate = propagate_log_messages
+    logging.getLogger("nslsii").propagate = propagate_log_messages
```

Comment:
It feels a for-loop for will have with the repeated patterns at least in 3 places above.